boston-crime-forecasting/content.md

```diff
@@ -276,27 +227,215 @@ for e in type_count.index:
+    correlation_df = col_by_col(col1, col2, batch_size, batch_size, lag, daily=daily)
+    x, y, z, coef_score = regression(correlation_df, regression_degree)
+    print('Y-axis: ', col2)
+    print('X-axis: ', col1)
+    print("Batch size: ", batch_size)
+    print(" ")
+    draw_regression(x, y, z)
+
+# Adjust this to examine data:
+print(weather_list)
+weather_list = ['Avg Temp (F)', 'High Temp (F)', 'Low Temp (F)']
+for i in range(len(weather_list)):
+    compare_cols('Avg Dew Point (F)', weather_list[i], batch_size=1)
+    compare_cols('Avg Dew Point (F)', weather_list[i], batch_size=10)
+
+weather_list = ['Avg Temp (F)', 'Avg Humidity (%)', 'Avg Sea Level Press (in)']
+for i in range(len(weather_list)):
+    compare_crime_weather(weather_list[i], 'Total', batch_size=1)
+    compare_crime_weather(weather_list[i], 'Total', batch_size=10)
+
+print(crime_list)
+crime_list = ['Warrant Arrests', 'Harassment']
+for i in range(len(crime_list)):
+    compare_cols(crime_list[i], 'Total', batch_size=1, daily=False, lag=1)
+    compare_cols(crime_list[i], 'Total', batch_size=10, daily=False, lag=1)
+    compare_cols(crime_list[i], 'Total', batch_size=50, daily=False, lag=1)
 ```
 
-## Forecasting with Recurrent Neural Network (RNN)
+Based on observations made in this section, we choose to ignore all crime categories except total crime. We will keep the weather
+categories `['Avg Temp (F)', 'Avg Humidity (%)', 'Avg Sea Level Press (in)', 'Avg Visibility (mi)', 'High Wind (mph)', 
+'Snowfall (in)', 'Precip (in)']`. We will also add columns for hour of the day, day of the week, day of the month and month 
+of the year.
+
+
+## Prepare Data for Recurrent Neural Network (RNN)
+Go back to the file 'prepare_data.py', and find the part at the end that looks like this:
+```python
+hourly_input = crime_by_hour.join(weather_data).fillna(method='ffill')
+daily_input = crime_by_day.join(weather_data).dropna()
+last_weather_hour = daily_input.iloc[-1].name + pd.to_timedelta('23 hours')
+hourly_input = hourly_input[:last_weather_hour]
+
+# save time series dataframes
+mflux_ai.init("hkPqTCtv-cUsROULi0Aizg")
+hourly_input.to_csv(os.path.join('data', 'hourly_data.csv'), index=False)
+mflux_ai.put_dataset(hourly_input, "hourly_input.pkl")
+daily_input.to_csv(os.path.join('data', 'daily_data.csv'), index=False)
+mflux_ai.put_dataset(daily_input, "daily_input.pkl")
+```
+
+We will replace part of this code and add to it. Simply replace that part with
+```python
+# based on data analysis
+crime_by_hour['Hour'] = crime_by_hour.index.hour
+crime_by_hour['Weekday'] = crime_by_hour.index.weekday
+crime_by_hour['Monthday'] = crime_by_hour.index.day
+crime_by_hour['Month'] = crime_by_hour.index.month
+crime_by_day['Weekday'] = crime_by_day.index.weekday
+crime_by_day['Monthday'] = crime_by_day.index.day
+crime_by_day['Month'] = crime_by_day.index.month
+weather_columns = ['Avg Temp (F)', 'Avg Humidity (%)', 'Avg Sea Level Press (in)', 'Avg Visibility (mi)', 'High Wind (mph)',
```

Comment:
This line is too long for the page width that we are using. I would recommend formatting the code with the `black` tool